library/ssl_msg.c

```diff
@@ -1820,7 +1820,7 @@ int mbedtls_ssl_fetch_input( mbedtls_ssl_context *ssl, size_t nb_want )
         {
             len = in_buf_len - ( ssl->in_hdr - ssl->in_buf );
 
-            if( ssl->state != MBEDTLS_SSL_HANDSHAKE_OVER )
+            if( !mbedtls_ssl_is_handshake_over( ssl ) )
```

Comment:
Style: Not sure if this is very important, but in my PRs I had requests to not use `!` inside `if` statement. Instead to use the following code style:
`if ( variable == value ) or if ( variable != value )`.

This comment applies also to other changes with the same problem (below).

Dealing with style problems is difficult for me in this project. I don't understand why we are not using `astye` or other formatting tool as it was done in mbed os.

Reply:
Usually we check for `== value` or `!= value` indeed, but I thought there was an exception for functions whose name indicates a boolean. I checked the [coding standards](https://tls.mbed.org/kb/development/mbedtls-coding-standards) document, and it doesn't say anything like this. It has a paragraph about return value conventions that indicates that function whose names indicate a boolean have a different return value convention, but at the end it seems to suggest that `== 0` and `!= 0` tests should be used for such functions as well.

So yes, `== 0` is probably the most consistent here (though in that case I'd personally find `!` a bit more readable).

Reply:
Slightly confused, you want all the comparisons re-done as comparing to zero (rather than zero for false, one for true?)

Reply:
No, sorry, I was not requesting any change, mostly ranting. Long story short, I think both version are fine: the present version because it conforms to the letter of our coding standards, and I think the previous version was fine as well because it conformed to the spirit of our coding standards.